internal/core/command/v2/application/command.go

```diff
@@ -54,3 +60,55 @@ func CommandsByDeviceName(name string, dic *di.Container) (commands []dtos.CoreC
 	}
 	return commands, nil
 }
+
+// IssueGetCommandByName issues the specified get(read) command referenced by the command name to the device/sensor, also
+// referenced by name.
+func IssueGetCommandByName(deviceName string, commandName string, pushEvent string, returnEvent string, dic *di.Container) (event dtos.Event, err errors.EdgeX) {
+	if deviceName == "" {
+		return event, errors.NewCommonEdgeX(errors.KindContractInvalid, "device name cannot be empty", nil)
+	}
+
+	if commandName == "" {
+		return event, errors.NewCommonEdgeX(errors.KindContractInvalid, "command name cannot be empty", nil)
+	}
+
+	if pushEvent == "" {
+		return event, errors.NewCommonEdgeX(errors.KindContractInvalid, "pushEvent cannot be empty", nil)
+	}
+
+	if returnEvent == "" {
+		return event, errors.NewCommonEdgeX(errors.KindContractInvalid, "returnEvent cannot be empty", nil)
+	}
+
+	// retrieve device information through Metadata DeviceClient
+	dc := V2Container.MetadataDeviceClientFrom(dic.Get)
+	if dc == nil {
+		return event, errors.NewCommonEdgeX(errors.KindClientError, "nil MetadataDeviceClient returned", nil)
+	}
+	deviceResponse, err := dc.DeviceByName(context.Background(), deviceName)
+	if err != nil {
+		return event, errors.NewCommonEdgeXWrapper(err)
+	}
+
+	// retrieve device service information through Metadata DeviceClient
+	dsc := V2Container.MetadataDeviceServiceClientFrom(dic.Get)
+	if dsc == nil {
+		return event, errors.NewCommonEdgeX(errors.KindClientError, "nil MetadataDeviceServiceClient returned", nil)
+	}
+	deviceServiceResponse, err := dsc.DeviceServiceByName(context.Background(), deviceResponse.Device.ServiceName)
+	if err != nil {
+		return event, errors.NewCommonEdgeXWrapper(err)
+	}
+
+	// Issue command by passing the base address of device service into DeviceServiceCommandClient
+	dscc := V2Container.DeviceServiceCommandClientFrom(dic.Get)
```

Comment:
What does putting this Client in the container gain us over a factory method to create the Client which at this point could take the base address as a parameter. I.e. creating this client ahead of time doesn't buy anything and adds more code for dealing with it being in the DIC.

Reply:
Putting the client in the DIC could help to reuse the client without creating the client in every single func invocation here, and this is the way we use other client libraries in V2 API implementation.  Putting the client in the DIC could also make the application layer protocol agnostic without knowing what kind of client instance it shall create.  Moreover, putting the client in the DIC could also make the unit tests easier to implement, as we could easily create a Mock client into the DIC.